ubidots/ubidots_in.html

```diff
@@ -1,111 +1,608 @@
+  </p>
+
+  <p>
+      Refer to the <a href="https://ubidots.com/docs/api/mqtt.html" target="_blank">Ubidots documentation</a> for more information about the functionality of the MQTT broker.
+  </p>
+  <p>-----</p>
+  <p>
+    If you get the error: "TypeError: send is not a function", please run <code>npm update node-red</code> and reload Node-Red.
+  </p>
+
 </script>
 
 <script type="text/javascript">
-    RED.nodes.registerType('ubidots_in', {
-        category: 'input',
-        color : "#5996F7",
-        defaults : {
-            tier: { value: 'business' },
-            name : {value : ""},
-            token : {value : ""},
-            device_label : {value : ""},
-            label_device : {value: ""},
-            label_variable : {value: ""}
-        },
-        inputs : 0,
-        outputs : 1,
-
-        icon : "ubidots.png",
-        label : function() {
-            return this.name || this.to || "ubidots input";
-        },
-        labelStyle : function() {
-            return this.name ? "node_label_italic" : "";
+  RED.nodes.registerType("ubidots_in", {
+    category: "input",
+    color: "#5996F7",
+    defaults: {
+      tier: {
+        value: "business"
+      },
+      name: {
+        value: "",
+        required: true
+      },
+      token: {
+        value: "",
+        required: true
+      },
+      device_label: {
+        value: ""
+      },
+      tls_checkbox_in: { value: true },
+      custom_topic_checkbox: { value: false },
+      label_variable_1: {
+        value: "",
+        required: true
+      },
+      label_variable_2: {
+        value: ""
+      },
+      label_variable_3: {
+        value: ""
+      },
+      label_variable_4: {
+        value: ""
+      },
+      label_variable_5: {
+        value: ""
+      },
+      label_variable_6: {
+        value: ""
+      },
+      label_variable_7: {
+        value: ""
+      },
+      label_variable_8: {
+        value: ""
+      },
+      label_variable_9: {
+        value: ""
+      },
+      label_variable_10: {
+        value: ""
+      },
+      checkbox_variable_1_last_value: { value: true },
+      checkbox_variable_2_last_value: { value: true },
+      checkbox_variable_3_last_value: { value: true },
+      checkbox_variable_4_last_value: { value: true },
+      checkbox_variable_5_last_value: { value: true },
+      checkbox_variable_6_last_value: { value: true },
+      checkbox_variable_7_last_value: { value: true },
+      checkbox_variable_8_last_value: { value: true },
+      checkbox_variable_9_last_value: { value: true },
+      checkbox_variable_10_last_value: { value: true }
+    },
+    inputs: 0,
+    outputs: 1,
+
+    icon: "ubidots.png",
+    label: function() {
+      return this.name || this.to || "Ubidots in";
+    },
+    labelStyle: function() {
+      return this.name ? "node_label_italic" : "";
+    },
+    oneditcancel: function() {},
```

Comment:
Just to make sure, this is required? If you remove this property which has a noop function it still works OK?